### ISSUE TITLE
cant send messages or  get response when using Httpserver

### ISSUE THREAD

**Issue by priyanka1230** [ratchetphp/Ratchet#553]
how can i send a message from webpage to telnet client?also ,to specific resourceid

new HttpServer(
              	new WsServer(
                    new WebSocketController()
                    )
              	),
                    8080

when using this,
no response on telnet

**Comment by halfer** [comment]
Your question lacks any level of detail to make it fully answerable, but broadly I would implement two channels:

* Browser to server (HTTP or WebSocket)
* Server to telnet

Your first communication channel uses standard protocols that browsers can use, i.e. AJAX or Web Sockets. If you wish to set up a live terminal then it'll probably need to be Web Sockets.

From there you can talk to a telnet server. You'll need to look on GitHub for a PHP library that can connect to and converse with telnet.

**Comment by priyanka1230** [comment]
when i use httpserver in my command file  with ioserver, i cant connect using telnet.whatever message i send from a telnet client or browser is not received by another client .


i want send a message to specific connected telnet client by selecting a client from dropdown which has all connectd clients.

i dont know how do i get list of all active telnet clients when sending request by ajax.

**Comment by halfer** [comment]
> [When I] use httpserver in my command file with ioserver, [I can't] connect using telnet

Why not? Be specific please - what problem do you specifically encounter?

**Comment by priyanka1230** [comment]
public function handle()
    {       
 $loop = IoServer::factory(  
                    new WebSocketController(),         
                    8081
          
);
 
$loop->run();

    }


this is my command file code.
i open two telnet connections
using,   telnet localhost 8081



then i open browser 
and from there send an ajax request to controller with parameters message and  resourceid of connected  telnet client.
i want to send message to that client.i dont know how to do that.

In short,i want to send message from browser to telnet terminals.how can i do that?what library should i use?i am using ratchet

**Comment by halfer** [comment]
@priyanka1230: please try to use block formatting for your code, so that it is as readable as possible. Use the preview feature if you are not sure what it will look like.

I searched [and found this telnet demo](https://stackoverflow.com/questions/22669467/reactphp-socket-server-what-triggers-the-write-to-client). I originally thought you would need a specific telnet-capable library, but it seems not. Does that help?

**Comment by priyanka1230** [comment]
@halfer 
Thanks for your support.I got the solution.By creating a new connection using fsockopen("127.0.0.1", 8081); on ajax call,
i was able to access all telnet connections and send message to particular client
